routing/graph.go

```diff
@@ -86,3 +92,38 @@ func (g *CachedGraph) fetchNodeFeatures(nodePub route.Vertex) (
 
 	return g.graph.FetchNodeFeatures(nodePub)
 }
+
+// FetchAmountPairCapacity determines the maximal capacity between two nodes.
+//
+// NOTE: Part of the routingGraph interface.
+func (g *CachedGraph) FetchAmountPairCapacity(nodeFrom, nodeTo route.Vertex,
+	amount lnwire.MilliSatoshi) (btcutil.Amount, error) {
+
+	// For the local node we assume no information on the channel capacity.
+	if nodeFrom == g.sourceNode() {
+		return 0, nil
+	}
+
+	// Create unified policies for all incoming connections.
+	u := newUnifiedPolicies(g.source, nodeTo, nil)
+
+	err := u.addGraphPolicies(g)
+	if err != nil {
+		return 0, err
+	}
+
+	// We may not have all policies available to describe the hop between
+	// the nodes (in the case of hop hints), which is why we return 0 in
+	// this case.
```

Comment:
I think this comment requires slightly more explanation. Why is it necessary to return 0 in case of hop hints? Maybe the explanation belongs in a different layer, because isn't `CachedGraph` purely about caching the graph without taking into account higher level usage patterns that might involve policies obtained from other sources?

Reply:
Agree, it is not the right place to have it here. I added a fixup commit with a solution how this could be handled closer to the rpc level, please have a look. I'm not sure if this is the best approach of returning errors here.